pkg/proxy/config/config.go

```diff
@@ -61,6 +63,12 @@ type EndpointsHandler interface {
 	// OnEndpointsSynced is called once all the initial event handlers were
 	// called and the state is fully propagated to local cache.
 	OnEndpointsSynced()
+
+	// SupportsEndpointSlices returns false to indicate that kube-proxy should use
+	// Endpoints rather than EndpointSlices. (If this returns true, kube-proxy will
+	// abort, because you should not implement EndpointsHandler if you support
+	// EndpointSlices.)
+	SupportsEndpointSlices() bool
```

Comment:
shouldn't an interface cast be enough to tell if Slices are supported or not a la:

```
slicesHandler, implemented := proxier.(<slices handler interface>)
```

Reply:
no, we already know that the proxier has _some_ implementation of `config.EndpointSliceHandler`, because that's a required sub-interface of `proxy.Provider`. The problem is that we have no way to figure out whether it's a "real" implementation or just a dummy one to satisfy the `interface` type.

If we removed `config.EndpointSliceHandler` from `proxy.Provider`, then we would no longer have compile-time checking that the proxy providers were implementing the complete interface... Well, I guess we could add new `proxy.DeprecatedProvider` and `proxy.FullProvider` interfaces containing the right sub-interfaces, and then each implementation could do

    var _ &Proxier{} = proxy.FullProvider

or

    var _ &Proxier{} = proxy.DeprecatedProvider

Reply:
/lgtm

I think we should leave as-is. Improve on later on.
Thanks.